### ISSUE TITLE
Tracing back additive HGTs

### ISSUE THREAD

**Issue by sivico26** [david-schaller/AsymmeTree#4]
Hello there,

Thanks for developing `AsymmeTree`, it is a nice piece of software. 

I am developing tools to detect HGTs and want to validate them through simulation. After checking the literature, I am giving AsymmeTree a chance. 

I am interested in simulating the HGTs and their fate at the sequence level and genome evolution. Because then I will use the sequences to test my tool.

As you pointed out in your paper, `AsymmeTree` is not primarily designed for this purpose, so I work around some things to make it work.

The first challenge I find is to extract the information about how the HGT happened. Take the following gene tree as an example:

![image](https://github.com/david-schaller/AsymmeTree/assets/37163025/0de4c576-d0ab-464e-bf1e-2362418a178b)

Here, you can spot a single HGT where species `<14>` donated to species `<10>`. Furthermore, I can see the receptor nodes and descendants on the gene tree (32 and 33). What I have not figured out is knowing the donor on the gene tree. Since there is a duplication, it could be either the ancestor of 28 and 29 or the ancestor of  30 and 31. I tried to see if I could infer it using the `tstamp` or the `dist` attributes, but I think I don't fully understand them to do so (As far as I can tell, `tstamp` won't help me, but maybe `dist?`)

Is there a way to know which copy was the donor in the event? This is important to me because I will have to recompose the Genomes out of the evolved sequences of each gene tree, and I need to know the donor copies to assign them properly. 

I have the suspicion (maybe reading your paper) that it is not possible by how the simulation is done (sampled from the species tree?), but I wanted to confirm this. I hope I am wrong. 

Thanks in advance

**Comment by AliLopSan** [comment]
Hello! 

Thanks for using our tool, we are glad to help `:)`

I've prepared a small example to illustrate how you can trace back the donor when an additive HGT event happens. Nodes in the gene tree contain an attribute called `reconc` which lets you trace back the donor edge in the species trees. This attribute is in fact a reconciliation map between the nodes of the gene tree and the nodes and edges of the species tree.

Suppose that you have a species tree whose newick string representation is:
> ((4,5)3,2)1;

And suppose we simulated a gene tree that looks like this:
![tree](https://github.com/david-schaller/AsymmeTree/assets/20115213/fedf3088-8171-4e08-9f27-2908ed43d9ad)

If you print all the nodes in this tree, their respective children nodes and the `event` and `reconc` attributes then we get the following:
|    n   | `n.event` | children | `n.reconc` |
|:------:|:---------:|----------|------------|
|  **0** |     S     | [1,2]    | 1          |
|  **1** |     S     | [3,4]    | 3          |
|  **3** |     H     | [5,6]    | (3,4)      |
|  **5** |     D     | [9,10]   | (3,4)      |
|  **9** |     D     | [13,14]  | (3,4)      |
| **13** |     S     | []       | 4          |
| **14** |     S     | []       | 4          |
| **10** |     S     | []       | 4          |
|  **6** |     S     | []       | 5          |
|  **4** |     D     | [7,8]    | (3,5)      |
|  **7** |     H     | [11,12]  | (3,5)      |
| **11** |     L     | []       | (3,5)      |
| **12** |     H     | [15,16]  | (3,4)      |
| **15** |     S     | []       | 4          |
| **16** |     S     | []       | 2          |
|  **8** |     S     | []       | 5          |
|  **2** |     S     | []       | 2          |

Note that the `reconc` attribute for HGT, loss and duplication nodes, i.e. those whose event is an `H` , `L` or `D` is a tuple, which belongs to the species tree edge in which the event took place. In the case of `H` it belongs to the donor edge in the species tree.

Let's suppose we take the node `7` of the gene tree, which is an HGT event. If you look at its parent node in the gene tree, then it would be node 4, which is a duplication node. By looking at the `reconc` attribute of `7`, we know that this event took place in the edge `(3,5)` of the species tree.

In your attached example, note that the HGT node has a tuple on its label `(5,10)`, so this means that the donor edge in the species tree was this one and not species `<14>` as you mention. In fact the red color of the edge indicates which node was transfered. In the example we attached, see that the transfer node `3` that happened on edge `(3,4)` of the species tree sent its children `6` to species `5`. You can verify which children of an `H` node was transfered through the `transferred` attribute.

We hope you find this answer useful, if you have any further questions please let us know,

**Comment by sivico26** [comment]
Hi @AliLopSan, 

Thanks for the quick reply and help. I am glad to confirm many how the aspects I inferred about the attributes of the gene trees. With your explanation, I realized that my conceptual mistake was to think that the edge defined by `.reconc` in the `H` nodes was the recipient node instead of the donor, which made me wonder about the donor copy. Now with this clarification looking for such a donor copy makes no sense (because it is the immediate ancestor).

Thanks a lot for the explanation.
Sivico
